fs/ensure_dir_test.ts

```diff
@@ -111,3 +113,45 @@ test(function ensureDirSyncIfItAsFile(): void {
 
   Deno.removeSync(baseDir, { recursive: true });
 });
+
+test(async function ensureDirIfItAsSymlink(): Promise<void> {
+  const testFile = path.join(testdataDir, "0-link.ts");
+
+  const testStat = await Deno.lstat(testFile);
+
+  if (isWindows) {
+    assert(testStat.isFile());
+    return;
+  } else {
+    assert(testStat.isSymlink());
+  }
+
+  await assertThrowsAsync(
+    async (): Promise<void> => {
+      await ensureDir(testFile);
+    },
+    Error,
+    `Ensure path exists, expected 'dir', got 'symlink'`
+  );
+});
+
+test(function ensureDirSyncIfItAsSymlink(): void {
+  const testFile = path.join(testdataDir, "0-link.ts");
+
+  const testStat = Deno.lstatSync(testFile);
+
+  if (isWindows) {
+    assert(testStat.isFile());
+    return;
+  } else {
+    assert(testStat.isSymlink());
+  }
+
+  assertThrows(
+    (): void => {
+      ensureDirSync(testFile);
```

Comment:
Here you're writing to the testdata dir?

Reply:
yes. I didn't update this PR yet.

Reply:
Next I will change the test case of the fs module one by one in another PR